src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AddHandler.scala

```diff
@@ -53,4 +53,69 @@ class AddHandler(authorizationProvider: AuthorizationProvider) extends Actor {
     val method = agoraBusiness.insert(agoraEntity.copy(createDate = Option(new DateTime())), username)
     context.parent ! RequestComplete(spray.http.StatusCodes.Created.intValue, method)
   }
+
+}
+
+case class MethodImportResolver(username: String, business: AgoraBusiness, authorizationProvider: AuthorizationProvider) {
+
+  def importResolver(importUri: String): WdlSource = {
+    ImportResolverHelper.validateUri(importUri)
+    val method = ImportResolverHelper.resolve(importUri, business, username)
+    val authorizedMethod = authorizationProvider.filterByReadPermissions(method, username)
```

Comment:
Is there not a "has read permission" method for a single entity? Seems weird to be filtering on a single entity instead of doing a boolean check.